### DISCUSSION TITLE
AOT: dotnet-setup.sh fails to fully execite (python missing using WSL)

### DISCUSSION THREAD

**Discussion by assassin316** [unoplatform/uno#14387] [General]
Executing the `dotnet-setup.sh` script to install emscripten in the WSL command prompt almost fully completes but fails with an error related to python:

![image](https://github.com/unoplatform/uno/assets/17514380/cca3a615-8150-4d6a-8e2c-be90b2275380)

I attempted to fix it by installing `python2`, `python-is-python3`, but nothing worked

**Comment by assassin316** [comment]
UPDATE:

May be linked to this:

https://github.com/unoplatform/Uno.Wasm.Bootstrap/issues/709

**Reply by jeromelaban** [reply]
Thanks for the report. What's prompting you to execute this particular script? Are you trying to build under Linux? WSL is not required to build wasm apps anymore in .NET 8.

**Reply by assassin316** [reply]
I'm still on .Net 5.0 and looking to upgrade in the near future. I'm executing this script for emscripten to be able to compile in InterperterAndAOT

**Reply by jeromelaban** [reply]
Thanks. To be sure, which version of the bootstrapper are you using?

**Reply by assassin316** [reply]
I'm using version **7.0.31**  (and 8.0.3 in my test using a .Net 7.0 Uno app)

Okay, so let me explain the issue and give you some more context.

Currently my app is built using Uno and UWP with .Net 5.0 (I am currently also migrating to 7.0 on my spare time).

My goal is to compile the Wasm project in release mode using InterpreterAndAOT.

According to the Uno documentation ([here](https://platform.uno/blog/build-net-aot-for-webassembly-in-visual-studio-with-uno-platform/)), I have to execute that script but since it doesn't work in PowerShell,  I use WSL to do this to be able to load Emscripten.

Here are my steps:

Step 1:  Building in Wasm Release mode gives me the following error:

![image](https://github.com/unoplatform/uno/assets/17514380/568793a4-269e-443c-aed2-f28158a4516e)

Step 2: Trying to install using the bash script gives errors with Python  (_the script targets .net 5.0 sdk_):

![image](https://github.com/unoplatform/uno/assets/17514380/447b9fcf-7ff4-4dc6-8477-08b5d18b2f4c)

Step 3: Trying to install using the bash script without Python gives other errors (_the script targets .net 5.0 sdk_):

![image](https://github.com/unoplatform/uno/assets/17514380/a055ec7f-35bb-4009-84b9-780946a8b155)

**HOWEVER**, I tried executing the bash script installing .Net 7.0 without Python in WSL and that completed successfully. **BUT**, even with a successful installation in WSL, the same error still occurs when build Wasm in Release mode:

![image](https://github.com/unoplatform/uno/assets/17514380/bbb6a650-4b04-407d-9247-b8931e4d71e6)

So basically, whether I'm doing with .Net 5.0 or .Net 7.0, I can't get the AOT build to execute successfully in Wasm due to the same Emscripten error. I've had this issue for a very long time, I never succeeded and kept pushing it, but now I'm at the point where I need to figure it out.

Thanks for your help. 👍

**Reply by jeromelaban** [reply]
Thanks for the update. Can you still provide a binlog? There's probably another error.

**Reply by assassin316** [reply]
Not sure if this is the culprit, but I found this in the log:

![image](https://github.com/unoplatform/uno/assets/17514380/e5e48cc9-656c-41ef-b4d5-edb1825646c6)

[binlog.zip](https://github.com/unoplatform/uno/files/13358326/binlog.zip)

**Reply by jeromelaban** [reply]
Thanks for the update. It seems that your csproj contains an invalid option, as seeing this command line:

```
[00:00:04.4527746] powershell "$env:PYTHONUTF8=1; emcc  -DENABLE_METADATA_UPDATE=1 -s USE_ZLIB=1 -s FORCE_FILESYSTEM=1  -fwasm-exceptions -DGEN_PINVOKE -IC:/Users/assas/AppData/Local/Temp/dotnet-runtime-wasm-windows-585dcd7-7db1c333330-4019238418-Release/runtimes/browser-wasm\native -emit-llvm  -lidbfs.js -l idbfs.js   -IC:/Users/assas/AppData/Local/Temp/dotnet-runtime-wasm-windows-585dcd7-7db1c333330-4019238418-Release/runtimes/browser-wasm/native/include/mono-2.0 -Oz -c -o corebindings.o corebindings.c"
```

Specifically `-lidbfs.js -l idbfs.js`, which seems to indicate `-l idbfs.js` which should be removed from your csproj as it's automatically added by the uno bootstrapper.

**Reply by assassin316** [reply]
As well, on a side note, the docs might need a small update:

https://platform.uno/blog/build-net-aot-for-webassembly-in-visual-studio-with-uno-platform/

Specifically here so that others don't encounter this issue like I did, Step 6 seems to be no longer needed.

![image](https://github.com/unoplatform/uno/assets/17514380/c669a392-b5a7-42ff-b934-b953c1bd0ff8)

**Reply by jeromelaban** [reply]
I see. 

This particular article very old, and is not in the docs. The doc is mostly talking about AOT compilation here: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html#runtime-execution-modes.

We can certainly link to more recent docs instead.

**Comment by assassin316** [comment]
It worked!! Thank you!

So the AOT build seems to be working as intended 👍

My last question regarding this is related to memory:

![image](https://github.com/unoplatform/uno/assets/17514380/2186e999-426c-487d-8518-65ccb294072a)

Where can I increase the memory for emcc?

According to the link below, there is a flag that can be specified: -s TOTAL_MEMORY=<my value> but since the uno bootstrapper adds the flags automatically, can we override it?

https://stackoverflow.com/questions/63595167/wasm-ld-error-initial-memory-too-small-18317952-bytes-needed

**Reply by jeromelaban** [reply]
Thanks for the update!

You can use [this option](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-module-linking.html#static-linking-additional-emcc-flags) to add the missing flag.

**Reply by assassin316** [reply]
Worked!
